tools/monitoring/landsat_gap_filler.py

```diff
@@ -1 +1,198 @@
+
+    # Post the last messages if there are any
+    if len(messages) > 0:
+        sent += len(messages)
+        publish_messages(queue, messages)
+
+    msg = f"Total messages sent {sent}"
+    if failed > 0:
+        msg = f":red_circle: Total of {failed} files failed, Total of sent messages {sent}"
+        if log:
+            log.error(f"{set(error_list)}")
+
+    return {"msg": msg, "fail": failed > 0}
+
+
+def build_message(missing_scene_paths, update_stac):
+    """ """
+    message_list = []
+    for path in missing_scene_paths:
+        landsat_product_id = str(path.strip("/").split("/")[-1])
+        if not landsat_product_id:
+            raise Exception(f"It was not possible to build product ID from path {path}")
+        message_list.append(
+            {
+                "Message": {
+                    "landsat_product_id": landsat_product_id,
+                    "s3_location": str(path),
+                    "update_stac": update_stac,
+                }
+            }
+        )
+    return message_list
+
+
+def fill_the_gap(
+    landsat: str,
+    sync_queue_name: str,
+    scenes_limit: Optional[int] = None,
+    notification_url: str = None,
+) -> None:
+    """
+    Function to retrieve the latest gap report and create messages to the filter queue process.
+
+    :param landsat:(str) satellite name
+    :param sync_queue_name:(str) Queue name
+    :param scenes_limit:(int) limit of how many scenes will be filled
+    :param notification_url:(str) Slack notification URL
+    :return:(None)
+    """
+    log = setup_logging()
+
+    try:
+
+        log.info(f"Satellite: {landsat}")
+        log.info(f"Queue: {sync_queue_name}")
+        log.info(f"Limited: {int(scenes_limit) if scenes_limit else 'No limit'}")
+        log.info(f"Notification URL: {notification_url}")
+
+        latest_report = find_latest_report(report_folder_path=S3_BUCKET_PATH)
+
+        if not latest_report:
+            logging.error("Report not found")
+            raise RuntimeError("Report not found!")
+
+        update_stac = False
+        if "update" in latest_report:
+            log.info("FORCED UPDATE FLAGGED!")
+            update_stac = True
+
+        log.info("Reading missing scenes from the report")
+
+        missing_scene_paths = read_report(report_path=latest_report, limit=scenes_limit)
+
+        log.info(f"Number of scenes found {len(missing_scene_paths)}")
+        log.info(f"Example scenes: {missing_scene_paths[0:10]}")
+
+        messages_to_send = build_message(
+            missing_scene_paths=missing_scene_paths, update_stac=update_stac
+        )
+
+        log.info("Publishing messages")
+        result = post_messages(
+            message_list=messages_to_send, queue_name=sync_queue_name, log=log
+        )
+
+        log.info(result["msg"])
+        if result["fail"]:
+            if slack_url is not None:
+                send_slack_notification(slack_url, "Landsat Gap Filler", result["msg"])
+            sys.exit(1)
```

Comment:
I don't think this function should be sys.exit(1). I think that should be done by the cli() function.

Reply:
If you check S2 gap report/filler, Landsat gap report, and check dead queues, none of them do **sys.exit(1)** on the **cli()** function. All of them do that in the main function which **cli()** calls.